Sources/Plasma/FeatureLib/pfPython/plPythonSDLModifier.cpp

```diff
@@ -314,6 +316,32 @@ void plPythonSDLModifier::IDirtySynchState(const ST::string& name, bool sendImme
     }
 }
 
+template<typename T>
+static std::optional<T> IConvertPythonNumber(PyObject* o)
+{
+    if constexpr (std::is_same_v<T, int>) {
+        if (PyLong_Check(o))
+            return PyLong_AsLong(o);
+        if (PyNumber_Check(o)) {
+            pyObjectRef pyLong = PyNumber_Long(o);
+            return PyLong_AsLong(pyLong.Get());
+        }
+        return std::nullopt;
+    } else if constexpr (std::disjunction_v<std::is_same<T, float>, std::is_same<T, double>>) {
+        if (PyFloat_Check(o))
+            return static_cast<T>(PyFloat_AS_DOUBLE(o));
+        if (PyNumber_Check(o)) {
+            pyObjectRef pyFloat = PyNumber_Float(o);
+            // pyFloat might have come from some strange land where they return
+            // unexpected things, so don't use the unsafe macro here.
+            return static_cast<T>(PyFloat_AsDouble(pyFloat.Get()));
+        }
+        return std::nullopt;
+    } else {
+        static_assert(false, "Unhandled C++ type conversion for Python number");
```

Comment:
This is not a dependent expression, and therefore is ill-formed even if the function is never called with the `else` condition ("no diagnostic required" means that MSVC is allowed to ignore it even though it's invalid, but GCC and Clang do emit a diagnostic)...  Some options:
1) Leave the base template unimplemented and implement the two (well, three) cases as specializations instead
2) Redefine the static_assert to use a dependent condition (something like `static_assert(sizeof(T) != sizeof(T), ...)`). Although this is technically still ill-formed, GCC and clang might be happier.
3) Move the static_assert outside of the `if constexpr` block, testing all possible types, and then leave something like a `std::unreachable()` in the else condition (less DRY).